components/scream/tests/coupled/dynamics_physics/homme_shoc_cld_p3_rrtmgp/CMakeLists.txt

```diff
@@ -6,7 +6,7 @@ CreateDynamicsLib("theta-l_kokkos"  4   72   35)
 
 # Create the test
 SET (TEST_LABELS "dynamics;driver;shoc;cld;p3;rrtmgp;physics")
-set (NEED_LIBS cld_fraction shoc p3 scream_rrtmgp ${dynLibName} scream_control scream_share physics_share)
+set (NEED_LIBS cld_fraction shoc spa p3 scream_rrtmgp ${dynLibName} scream_control scream_share physics_share)
```

Comment:
Minor, but you do not need the spa library. Not a big deal, but during dev cycles, this dep can trigger rebuild of spa, even if I'm just trying to run this test.

Reply:
we actually do, if we use `register_physics.hpp` in the main file.  Because register_physics still registers SPA even if it's not used.

Reply:
Ugh, then we should not call register_phyisics, and rather do the registration manually.